spec/mailers/candidate_mailer_offers_and_rejections_spec.rb

```diff
@@ -0,0 +1,193 @@
+require 'rails_helper'
+
+RSpec.describe CandidateMailer, type: :mailer do
+  include CourseOptionHelpers
+  include ViewHelper
+  include TestHelpers::MailerSetupHelper
+
+  subject(:mailer) { described_class }
+
+  shared_examples 'a mail with subject and content' do |mail, subject, content|
+    let(:email) { described_class.send(mail, @application_choice) }
+
+    it 'sends an email with the correct subject' do
+      expect(email.subject).to include(subject)
+    end
+
+    content.each do |key, expectation|
+      it "sends an email containing the #{key} in the body" do
+        expect(email.body).to include(expectation)
+      end
+    end
+  end
+
+  before do
+    setup_application
+  end
+
+  describe 'send new offer email to candidate' do
+    around do |example|
+      Timecop.freeze(Time.zone.local(2020, 2, 11)) do
+        example.run
+      end
+    end
+
+    describe '#new_offer_single_offer' do
+      it_behaves_like(
+        'a mail with subject and content', :new_offer_single_offer,
+        'Offer received for Applied Science (Psychology) (3TT5) at Brighthurst Technical College',
+        'heading' => 'Dear Bob',
+        'decline by default date' => 'Make a decision by 25 February 2020',
+        'first_condition' => 'DBS check',
+        'second_condition' => 'Pass exams'
+      )
+    end
+
+    describe '#new_offer_multiple_offers' do
+      before do
+        provider = build_stubbed(:provider, name: 'Falconholt Technical College')
+        other_course_option = build_stubbed(:course_option, course: build_stubbed(:course, name: 'Forensic Science', code: 'E0FO', provider: provider))
+        @other_application_choice = @application_form.application_choices.build(
+          id: 456,
```

Comment:
Is the setting of `id` needed?

Reply:
No, I have no idea why did we have them 🤔 cut ✂️